lib/galaxy/tool_util/provided_metadata.py

```diff
@@ -53,6 +53,8 @@ def get_new_datasets(self, output_name):
 
     def has_failed_outputs(self):
         """Determine if generation of any of the outputs failed.
+
+        If True, this method should also log information about at the first such failed output.
```

Comment:
```suggestion
        If True, this method should also log information about such failed outputs.
```

Reply:
I meant for this to be "at least the first such failed output." - do you still prefer what you have here? This doesn't log about each failed outputs with these changes, only the first.

Reply:
Noe, yours is clear. I wasn't sure what you wanted here.